Feature/doc errores

pepesouto opened the pull request:
# PR Template

## Descripción

Se agrego al readme los errores de cada funcionalidad.

pepesouto committed:
userInfo errores agregados

pepesouto committed:
logout errores

belenbrandino commented on sdk/README.md:
```diff
@@ -266,6 +266,14 @@ initialize(
 
 Luego de esto, se considera que el SDK se encuentra inicializado correctamente.
 
+Si alguno de estos parámetros obligatorios es vacío devuelve un error correspondiente a cual es el primero vacío.
```
pondría devuelve un error indicando cual es el primer parámetro vacío

belenbrandino commented on sdk/README.md:
```diff
@@ -266,6 +266,14 @@ initialize(
 
+Si alguno de estos parámetros obligatorios es vacío devuelve un error correspondiente a cual es el primero vacío.
+
+*clientId* vacío retorna `ERRORS.INVALID_CLIENT_ID`.
```
pondría, si el parámetro *clientId* es vacío retorna ...

belenbrandino commented on sdk/README.md:
```diff
@@ -266,6 +266,14 @@ initialize(
+*redirectUri* vacía retorna `ERRORS.INVALID_REDIRECT_URI`.
+*clientSecret* vacío retorna `ERRORS.INVALID_CLIENT_SECRET`.
+
+Por otro lado, si el parámetro production no es booleano se retorna el error `ERRORS.INVALID_PRODUCTION`.
```
pondría: si **el tipo del** parámetro production no es booleano, se retorna

belenbrandino commented on sdk/README.md:
```diff
@@ -291,6 +299,14 @@ try {
 
 Se debe notar que si el usuario final no inicia la sesión con ID Uruguay (ya sea porque cierra el navegador, o porque ingresa credenciales incorrectas), no se redirigirá a la *redirect_uri* especificada.
 
+En caso de que alguno de los parámetros *clientId*, *clientSecret* y *redirectUri* no haya sido seteado, por lo tanto sea vacío, se retorna el error correspondiente al primer parámetro vacio: `ERRORS.INVALID_CLIENT_ID`, `ERRORS.INVALID_REDIRECT_URI`, `ERRORS.INVALID_CLIENT_SECRET`.
```
Unificaría el formato con el de los errores del login (o lista o texto). Si decidís usar texto, los pondría en el mismo orden, es decir si arriba decís *clientId*, *clientSecret* y *redirectUri* abajo pondría 
correspondiente al primer parámetro vacio, siendo estos: `ERRORS.INVALID_CLIENT_ID`, `ERRORS.INVALID_CLIENT_SECRET` y `ERRORS.INVALID_REDIRECT_URI` respectivamente

belenbrandino replied:
Lo mismo para todos, deberían tener el mismo formato para describir los errores

belenbrandino replied:
debería mencionar tmb el error de producción (revisar esto, si va o no)

belenbrandino commented on sdk/README.md:
```diff
@@ -291,6 +299,14 @@ try {
 
+En caso de que alguno de los parámetros *clientId*, *clientSecret* y *redirectUri* no haya sido seteado, por lo tanto sea vacío, se retorna el error correspondiente al primer parámetro vacio: `ERRORS.INVALID_CLIENT_ID`, `ERRORS.INVALID_REDIRECT_URI`, `ERRORS.INVALID_CLIENT_SECRET`.
+      
+En caso de que no exista el parámetro *code* en la url se retorna el error `ERRORS.INVALID_AUTHORIZATION_CODE`.
```
en la url retornada por el OP?

belenbrandino commented on sdk/README.md:
```diff
@@ -291,6 +299,14 @@ try {
+      
+En caso de que no exista el parámetro *code* en la url se retorna el error `ERRORS.INVALID_AUTHORIZATION_CODE`.
+
+En caso de que el usuario niegue dar acceso de la información solicitada se retorna el error `ERRORS.ACCESS_DENIED`.
```
lo escribiría un poco más lindo, además de que debería decir usuario **final**. Algo como lo que dice la arqui: _El usuario final no autoriza a la aplicación móvil RP a acceder a sus datos_

belenbrandino commented on sdk/README.md:
```diff
@@ -291,6 +299,14 @@ try {
+
+En caso de que el usuario niegue dar acceso de la información solicitada se retorna el error `ERRORS.ACCESS_DENIED`.
+
+En caso de error en la request se retorna `ERRORS.FAILED_REQUEST`, en este caso se recomienda revisar que los parámetros de la request sean válidos y comprobar la conexión a internet.
```
request con cursiva no?

belenbrandino commented on sdk/README.md:
```diff
@@ -301,6 +317,14 @@ const token = await getToken();
 
 Al igual que el `code`, el *token* retornado se guarda en el SDK, con lo que de no necesitar almacenar el *token*, también se puede llamar a `getToken` sin guardar la respuesta.
 
+En caso de que alguno de los parámetros *clientId*, *clientSecret*, *redirectUri* y *code* no haya sido seteado, por lo tanto sea vacío, se retorna el error correspondiente al primer parámetro vacio: `ERRORS.INVALID_CLIENT_ID`, `ERRORS.INVALID_REDIRECT_URI`, `ERRORS.INVALID_CLIENT_SECRET` y  `ERRORS.INVALID_AUTHORIZATION_CODE`.
```
poner ordenados los errores y lo de respectivamente

belenbrandino commented on sdk/README.md:
```diff
@@ -301,6 +317,14 @@ const token = await getToken();
 
+En caso de que alguno de los parámetros *clientId*, *clientSecret*, *redirectUri* y *code* no haya sido seteado, por lo tanto sea vacío, se retorna el error correspondiente al primer parámetro vacio: `ERRORS.INVALID_CLIENT_ID`, `ERRORS.INVALID_REDIRECT_URI`, `ERRORS.INVALID_CLIENT_SECRET` y  `ERRORS.INVALID_AUTHORIZATION_CODE`.
+
+En caso de que el *code* sea inválido o haya expirado, y no se puede obtener un nuevo token de forma satisfactoria se retorna `ERRORS.INVALID_GRANT`.
```
y no se pued**A** obtener un token? token con cursiva. También puede ser vacío el code y retornar el mismo error

pepesouto replied:
si es vacio el code no retorna `ERRORS.INVALID_AUTHORIZATION_CODE` ?

belenbrandino commented on sdk/README.md:
```diff
@@ -301,6 +317,14 @@ const token = await getToken();
+
+En caso de que el *code* sea inválido o haya expirado, y no se puede obtener un nuevo token de forma satisfactoria se retorna `ERRORS.INVALID_GRANT`.
+
+En caso de que no se pudo obtener un nuevo token de forma correcta se retorna `ERRORS.INVALID_CLIENT`. Se recomienda verificar que el *client_secret* y *client_id* correspondan con los registrados por ID Uruguay. 
```
Este error se retorna en caso de que pongas un client id o client secret no registrado ante el OP

belenbrandino commented on sdk/README.md:
```diff
@@ -301,6 +317,14 @@ const token = await getToken();
+
+En caso de que no se pudo obtener un nuevo token de forma correcta se retorna `ERRORS.INVALID_CLIENT`. Se recomienda verificar que el *client_secret* y *client_id* correspondan con los registrados por ID Uruguay. 
+
+En caso de error en la respuesta del endpoint se retorna `ERRORS.FAILED_REQUEST`.
```
Les pondría a todas las funciones la misma justificación. Failed request es el ultimo error que tiras dsp de verificar todos los q conoces, osea, si te viene un error desconocido (no controlado), sale por acá

belenbrandino commented on sdk/README.md:
```diff
@@ -311,6 +335,18 @@ const token = await refreshToken();
 
+Los casos de errores son muy similares a los de `getToken`. 
+
+En caso de que alguno de los parámetros *clientId*, *clientSecret* y *redirectUri* sea vacío, se retorna el error correspondiente al primer parámetro vacio: `ERRORS.INVALID_CLIENT_ID`, `ERRORS.INVALID_REDIRECT_URI`, `ERRORS.INVALID_CLIENT_SECRET`.
```
orden

belenbrandino commented on sdk/README.md:
```diff
@@ -311,6 +335,18 @@ const token = await refreshToken();
 
 Esta función requiere que la función `getToken` haya sido ejecutada de forma correcta.
 
+Los casos de errores son muy similares a los de `getToken`. 
```
de la funcionalidad getToken?

belenbrandino commented on sdk/README.md:
```diff
@@ -311,6 +335,18 @@ const token = await refreshToken();
+
+En caso de que alguno de los parámetros *clientId*, *clientSecret* y *redirectUri* sea vacío, se retorna el error correspondiente al primer parámetro vacio: `ERRORS.INVALID_CLIENT_ID`, `ERRORS.INVALID_REDIRECT_URI`, `ERRORS.INVALID_CLIENT_SECRET`.
+
+A diferencia del `getToken`, no se chequea que el code sea vacio, pero en cambio se revisa que que el parámetro `refreshToken` no lo sea. Si refreshToken es vacio se retorna `ERRORS.INVALID_GRANT`. 
```
A diferencia de la funcionalidad getToken? Code con cursiva. Pondría solo ", en cambio se revisa". Al final pondría, en tal caso se retorna ... 
Y vacío lleva tilde

belenbrandino commented on sdk/README.md:
```diff
@@ -311,6 +335,18 @@ const token = await refreshToken();
+
+A diferencia del `getToken`, no se chequea que el code sea vacio, pero en cambio se revisa que que el parámetro `refreshToken` no lo sea. Si refreshToken es vacio se retorna `ERRORS.INVALID_GRANT`. 
+
+En caso de que el refreshToken sea inválido o haya expirado se retorna `ERRORS.INVALID_GRANT`.
```
estamos seguros que el refreshToken expira?

belenbrandino commented on sdk/README.md:
```diff
@@ -311,6 +335,18 @@ const token = await refreshToken();
+
+En caso de que el refreshToken sea inválido o haya expirado se retorna `ERRORS.INVALID_GRANT`.
+
+En caso de que no se pudo obtener un nuevo token de forma correcta se retorna `ERRORS.INVALID_CLIENT`. Se recomienda verificar que el *client_secret* y *client_id* correspondan con los registrados por ID Uruguay. 
```
lo mismo que en getToken

belenbrandino commented on sdk/README.md:
```diff
@@ -311,6 +335,18 @@ const token = await refreshToken();
+
+En caso de que no se pudo obtener un nuevo token de forma correcta se retorna `ERRORS.INVALID_CLIENT`. Se recomienda verificar que el *client_secret* y *client_id* correspondan con los registrados por ID Uruguay. 
+
+En caso de error en la respuesta endpoint se retorna `ERRORS.FAILED_REQUEST`.
```
unificar

belenbrandino commented on sdk/README.md:
```diff
@@ -266,6 +266,14 @@ initialize(
 
 Luego de esto, se considera que el SDK se encuentra inicializado correctamente.
 
```
en todos los casos deberíamos decir que cuando no hay error retornamos el no error

belenbrandino commented on sdk/README.md:
````diff
@@ -331,6 +367,10 @@ Esta función devuelve un objeto con el siguiente formato:
 }
 ```
 
+En caso de que el *accessToken* sea inválido devuelve un error de tipo `ERRORS.INVALID_TOKEN`.
````
En caso de que el parámetro ... Además, esto pasa cuando es vacío también

belenbrandino commented on sdk/README.md:
````diff
@@ -331,6 +367,10 @@ Esta función devuelve un objeto con el siguiente formato:
 }
 ```
````
falta cuando id token es vacío y cuando el sub es inválido

belenbrandino commented on sdk/README.md:
```diff
@@ -301,6 +317,14 @@ const token = await getToken();
 
 Al igual que el `code`, el *token* retornado se guarda en el SDK, con lo que de no necesitar almacenar el *token*, también se puede llamar a `getToken` sin guardar la respuesta.
```
debería mencionar tmb el error de producción (revisar si va o no)

belenbrandino commented on sdk/README.md:
```diff
@@ -301,6 +317,14 @@ const token = await getToken();
+
 ### Función refreshToken
 
 El *token* otorgado por ID Uruguay tiene un tiempo de expiración fijo, por lo que una vez transcurrido este tiempo, el *token* pasará a ser inválido. Para obtener un nuevo *token* se debe invocar a la función `refreshToken`.
```
debería mencionar tmb el error de producción  (revisar si va o no)

belenbrandino commented on sdk/README.md:
```diff
@@ -331,6 +367,10 @@ Esta función devuelve un objeto con el siguiente formato:
 
+En caso de que el *accessToken* sea inválido devuelve un error de tipo `ERRORS.INVALID_TOKEN`.
+
+En caso de error en la respuesta del endpoint se retorna `ERRORS.FAILED_REQUEST`.
```
unificar

belenbrandino commented on sdk/README.md:
````diff
@@ -360,7 +400,14 @@ try {
 ```
 
-En caso de que el *token* sea inválido devuelve un error de tipo `ERRORS.INVALID_ID_TOKEN`.
+Si alguno de los parámetros obligatorios para la request no se encuentra inicializado, se rechaza la promesa y se retorna un error que especifica cuál parámetro faltó.
````
el primero que faltó

belenbrandino commented on sdk/README.md:
```diff
@@ -360,7 +400,14 @@ try {
+
+En caso de que el *clientId* sea vacío devuelve un error de tipo `ERRORS.INVALID_CLIENT_ID`.
+
+En caso de que el *token* no se pueda validar en el modulo de seguridad se retorna el error `ERRORS.FAILED_REQUEST`.
```
en ese caso devuelve INVALID_ID_TOKEN

belenbrandino commented on sdk/README.md:
```diff
@@ -360,7 +400,14 @@ try {
+En caso de que el *clientId* sea vacío devuelve un error de tipo `ERRORS.INVALID_CLIENT_ID`.
+
+En caso de que el *token* no se pueda validar en el modulo de seguridad se retorna el error `ERRORS.FAILED_REQUEST`.
+
```
falta el ultimo error de failed request

belenbrandino commented on sdk/README.md:
```diff
@@ -360,7 +400,14 @@ try {
+En caso de que el *token* no se pueda validar en el modulo de seguridad se retorna el error `ERRORS.FAILED_REQUEST`.
+
 
 ### Función logout
```
en los errores de logout habla de parámetros obligatorios: cuales? Y falta unificar lo de failed request. En la línea 437 está mal escrito logoutEndpoint. De paso modificar el logout.js en la línea 54 tiene el mismo typo

belenbrandino commented:
Faltan todos estos errores en contributing.md. Hablar con Leo sobre si esta bien que los errores estén definidos abajo del todo en sdk/readme y vos los menciones antes

belenbrandino reviewed (changes requested):
.

elirosselli reviewed (changes requested):
Holi :) Deberíamos cambiar las llamadas de ejemplo a las funciones para que todas usen try-catch como el login

pepesouto committed:
cambios bela, falta eli y production va?

guillermoG23 committed:
pequeños cambios 

Puse try-catch en los codigos de ejemplos.
Saque los campos name de las tablas de errores porque entendi que no iba mas ese campo.

pepesouto committed:
errores finished

pepesouto committed:
Merge branch 'develop' into feature/doc-errores

pepesouto committed:
merge fix

pepesouto committed:
minor fix

pepesouto committed:
typos

belenbrandino commented:
> Faltan todos estos errores en contributing.md. Hablar con Leo sobre si esta bien que los errores estén definidos abajo del todo en sdk/readme y vos los menciones antes

Todavía no lei las cosas nuevas, si siguen la semana que viene las leo :)

slagomarsino committed:
Fixed markdown linter

guillermoG23 committed:
arreglos menores

Eliminadas las funciones eraseCode y eraseState

guillermoG23 committed:
cambio minimo

slagomarsino committed:
Fix typo

slagomarsino committed:
Merge branch 'develop' into feature/doc-errores
